src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java

```diff
@@ -205,40 +202,36 @@ public ResponseEntity<Course> updateCourse(@RequestBody Course updatedCourse) th
         else {
             // this means the user must be an instructor, who has NO Admin rights.
             // instructors are not allowed to change group names, because this would lead to security problems
-
-            if (!Objects.equals(existingCourse.getStudentGroupName(), updatedCourse.getStudentGroupName())) {
-                throw new BadRequestAlertException("The student group name cannot be changed", Course.ENTITY_NAME, "studentGroupNameCannotChange", true);
-            }
-            if (!Objects.equals(existingCourse.getTeachingAssistantGroupName(), updatedCourse.getTeachingAssistantGroupName())) {
-                throw new BadRequestAlertException("The teaching assistant group name cannot be changed", Course.ENTITY_NAME, "teachingAssistantGroupNameCannotChange", true);
-            }
-            if (!Objects.equals(existingCourse.getEditorGroupName(), updatedCourse.getEditorGroupName())) {
-                throw new BadRequestAlertException("The editor group name cannot be changed", Course.ENTITY_NAME, "editorGroupNameCannotChange", true);
-            }
-            if (!Objects.equals(existingCourse.getInstructorGroupName(), updatedCourse.getInstructorGroupName())) {
-                throw new BadRequestAlertException("The instructor group name cannot be changed", Course.ENTITY_NAME, "instructorGroupNameCannotChange", true);
+            if (!changedGroupNames.isEmpty()) {
+                throw new BadRequestAlertException("You are not allowed to change the group names of a course", Course.ENTITY_NAME, "groupNamesCannotChange", true);
             }
         }
 
         // Make sure to preserve associations in updated entity
-        updatedCourse.setPrerequisites(existingCourse.getPrerequisites());
-
-        updatedCourse.validateRegistrationConfirmationMessage();
-        updatedCourse.validateComplaintsAndRequestMoreFeedbackConfig();
-        updatedCourse.validateOnlineCourseAndRegistrationEnabled();
-        updatedCourse.validateShortName();
-        updatedCourse.validateAccuracyOfScores();
-        if (!updatedCourse.isValidStartAndEndDate()) {
+        courseUpdate.setPrerequisites(existingCourse.getPrerequisites());
+
+        courseUpdate.validateRegistrationConfirmationMessage();
+        courseUpdate.validateComplaintsAndRequestMoreFeedbackConfig();
+        courseUpdate.validateOnlineCourseAndRegistrationEnabled();
+        courseUpdate.validateShortName();
+        courseUpdate.validateAccuracyOfScores();
+        if (!courseUpdate.isValidStartAndEndDate()) {
             throw new BadRequestAlertException("For Courses, the start date has to be before the end date", Course.ENTITY_NAME, "invalidCourseStartDate", true);
         }
 
+        if (file != null) {
+            String pathString = fileService.handleSaveFile(file, false, false);
+            courseUpdate.setCourseIcon(pathString);
+        }
+
+        Course result = courseRepository.save(courseUpdate);
```

Comment:
Should we check that the id of course update is actually the id from the URL-path?